### DISCUSSION TITLE
compoents.classes property type define

### DISCUSSION THREAD

**Discussion by liaoyinglong** [mantinedev/mantine#6048] [Q&A] [answered]
First of all, I'd like to say that your library is great. I really appreciate the tooling and the docs. Excellent work! 

I wanted to mention that I noticed the `Button.classes` component isn't type-strong - currently, the type being used is `Record<string, string>`. I was wondering if it could be updated to use either `Record<'root' | 'inner' | 'loader' | 'section' | 'label', string>` or `Record<ButtonStylesNames, string>` instead?

**Comment by rtivital** [comment, marked as answer]
Thanks for kind words! `classes` are not typed strongly to avoid issues with different behavior of CSS modules imports in different environments. It is not planned to change this behavior.
